ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -105,6 +107,13 @@ else if (desc.getPartitionSpec() != null) {
       compactionRequest.setNumberOfBuckets(desc.getNumberOfBuckets());
     }
 
+    if (desc.getFileSizeThreshold() != null) {
+      if (!DDLUtils.isIcebergTable(table) || compactionRequest.getType() != CompactionType.MINOR) {
+        throw new HiveException(ErrorMsg.UNSUPPORTED_COMPACTION_REQUEST_WITH_FILE_SIZE_THRESHOLD);
```

Comment:
wait, it's not supported by Major compaction?

Reply:
Done